controllers/nutanixcluster_controller.go

```diff
@@ -170,20 +170,11 @@ func (r *NutanixClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		log.V(1).Info(fmt.Sprintf("Patched NutanixCluster. Status: %+v", cluster.Status))
 	}()
 
-	err = r.reconcileCredentialRef(ctx, cluster)
```

Comment:
What do you think about keeping `r.reconcileCredentialRef` but also returning the `Secret` and then passing that Secret into `getPrismCentralClientForCluster`, that would make the function a little more focused and more unit testable?

It would also make the changes in this PR smaller.

(originally left this comment in https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/398#discussion_r1575308135)

Reply:
as discussed offline, the `reconcileCredentialRef` is getting called from both machine and cluster controller and as such can't be a method on the cluster reconciler.

Reply:
With the responsibility of providing the secret contents management at any time shifted to the client cache itself, this no longer needs to be called by machine reconciler. I've moved it back to it's original role as a method on cluster reconciler.